### DISCUSSION TITLE
Single thread performance w.r.t openssl config options "threads vs no-threads"

### DISCUSSION THREAD

**Discussion by jsrikanth1608** [openssl/openssl#25332] [Q&A] [answered]
Hi,
What are the performance benefit for single threaded applications by configuring OpenSSL with "no-threads".? Does it makes all locking functions as no-op and save cycles of lock/unlock function calls.?

Thank you,
Srikanth

**Comment by paulidale** [comment, marked as answer]
Yes.  Locking calls become no-ops, however the function call doesn't always disappear.

**Reply by jsrikanth1608** [reply]
Thank you.
